### PR TITLE
v1.11 backports 2022-09-07

### PR DESCRIPTION
**Pull request by joamaki**
* #20112 -- Fix conflicting routes for multiple ENIs in IPAM mode (@recollir)
 * #21093 -- pkg/k8s/watcher: fix deadlock with service event handler & CES watcher. (@tommyp1ckles)
 * #20795 -- agent: Stop serving DNS before shutdown (@nebril)
 * #21131 -- metallb: bump to latest metallb version (@ldelossa)
 * #21133 -- operator: do not GC kvstore nodes if CiliumNodes are not available (@aanm)
 * #21146 -- datapath: tolerate missing ifaces when setting rp_filter sysctl (@julianwiedmann)
 * #21126 -- datapath: allow local NodePort traffic for `eni+` container interfaces with CNI chaining (@ti-mo)
 * #20848 -- Coalesce of health endpoint CIDRs (@dezmodue)
 * #21127 -- hubble-ui: release v0.9.2 (@geakstr)
 * #21202 -- operator: update CiliumNode in kvstore without lease (@tklauser)
 * #21220 -- k8s: fix test flake in TestGenerateToCIDRFromEndpoint. (@tommyp1ckles)

Once this PR is merged, you can update the PR labels via:
```upstream-prs
$ for pr in 20112 21093 20795 21131 21133 21146 21126 20848 21127 21202 21220; do contrib/backporting/set-labels.py $pr done 1.11; done
```
or with
```
$ make add-label BRANCH=v1.11 ISSUES=20112,21093,20795,21131,21133,21146,21126,20848,21127,21202,21220
```

### CODE REVIEW

**Review by julianwiedmann — APPROVED** [review]

**Review by tklauser — APPROVED** [review]
> * [operator: update CiliumNode in kvstore without lease #21202](https://github.com/cilium/cilium/pull/21202) -- operator: update CiliumNode in kvstore without lease (@tklauser)

Backport of that change looks good. Thanks.

**Review by geakstr — APPROVED** [review]

**Review by aanm — APPROVED** [review]

**Review by ldelossa — APPROVED** [review]

**Comment by tommyp1ckles** [comment]
@joamaki I recently merged: https://github.com/cilium/cilium/pull/21220 as well which contains a fix for a test flake introduced in: https://github.com/cilium/cilium/pull/21093

File: pkg/option/config.go [hunk 1]
```diff
@@ -2665,6 +2671,7 @@ func (c *DaemonConfig) Populate() {
 	c.KeepConfig = viper.GetBool(KeepConfig)
 	c.KVStore = viper.GetString(KVStore)
 	c.KVstoreLeaseTTL = viper.GetDuration(KVstoreLeaseTTL)
+	c.CNIChainingMode = viper.GetString(CNIChainingMode)
```
Comment (ti-mo) [inline pkg/option/config.go]: Same as https://github.com/cilium/cilium/pull/21225#discussion_r965588268.

**Comment by nebril** [comment]
/test-backport-1.11

**Review by nebril — APPROVED** [review]

**Comment by nebril** [comment]
/test-backport-1.11

**Review by tommyp1ckles — APPROVED** [review]

**Review by ti-mo — APPROVED** [review]
Thanks!

**Pull request merged**